quic/s2n-quic-core/src/datagram/default.rs

```diff
@@ -318,6 +353,12 @@ impl Sender {
             return Err(SendDatagramError::ExceedsPeerTransportLimits);
         }
 
+        // If there was some connection-level error the user is not allowed to add
+        // datagrams to the queue as they will never be sent.
+        if let Some(err) = self.error {
+            return Err(SendDatagramError::ConnectionError { error: err });
+        }
```

Comment:
```
if data.len() as u64 > self.max_datagram_payload {
...
if let Some(err) = self.error {
```
this check is repeated in 3 places. can we consolidate this check in a single function.